File: app/src/main/java/com/klinker/android/twitter_l/data/roomdb/QueuedTweetDao.java
package com.klinker.android.twitter_l.data.roomdb;


import java.util.List;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

@Dao
public interface QueuedTweetDao {

    @Insert
    void insertQueuedTweet(QueuedTweet queuedTweet);

    @Delete
    void deleteQueuedTweet(QueuedTweet queuedTweet);

    @Query("DELETE FROM queued_tweets WHERE text = :message AND account = :account")
    void deleteQueuedTweet(String message, int account);

    @Query("SELECT * FROM queued_tweets WHERE account = :account")
    List<QueuedTweet> getQueuedTweets(int account);


}

File: app/src/main/java/com/klinker/android/twitter_l/data/roomdb/daos/QueuedTweetDao.kt
package com.klinker.android.twitter_l.data.roomdb.daos


import com.klinker.android.twitter_l.data.roomdb.entities.QueuedTweet

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface QueuedTweetDao {

    @Insert
    fun insertQueuedTweet(queuedTweet: QueuedTweet) : Long

    @Delete
    fun deleteQueuedTweet(queuedTweet: QueuedTweet)

    @Query("SELECT * FROM queued_tweets WHERE account = :account")
    fun getQueuedTweets(account: Int): List<QueuedTweet>

    @Query("DELETE FROM queued_tweets")
    fun deleteAllQueuedTweets()


}
